com.unity.perception/Editor/Visualizer/VisualizerInstaller.cs

```diff
@@ -0,0 +1,765 @@
+        /// </summary>
+        /// <param name="pid"></param>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        static bool CheckProcessName(int pid, string name)
+        {
+            var proc = Process.GetProcessById(pid + 1);
+            return proc.ProcessName.ToLower().Contains(name);
+        }
+
+        /// <summary>
+        /// Check if the given PID listens to given port
+        /// </summary>
+        /// <param name="pid"></param>
+        /// <param name="port"></param>
+        /// <returns></returns>
+        static bool ProcessListensToPort(int pid, int port)
+        {
+            var processes = ProcessPorts.ProcessPortMap.FindAll(
+                x => x.ProcessId == pid + 1 && x.PortNumber == port
+            );
+            return processes.Count >= 1;
+        }
+
+        /// <summary>
+        /// Static class that returns the list of processes and the ports those processes use.
+        /// </summary>
+        static class ProcessPorts
+        {
+            /// <summary>
+            /// A list of ProcessPorts that contain the mapping of processes and the ports that the process uses.
+            /// </summary>
+            public static List<ProcessPort> ProcessPortMap => GetNetStatPorts();
+
+            /// <summary>
+            /// This method distills the output from Windows: netstat -a -n -o or OSX: netstat -v -a into a list of ProcessPorts that provide a mapping between
+            /// the process (name and id) and the ports that the process is using.
+            /// </summary>
+            /// <returns></returns>
+            static List<ProcessPort> GetNetStatPorts()
+            {
+                var processPorts = new List<ProcessPort>();
+
+                var startInfo = new ProcessStartInfo();
+#if UNITY_EDITOR_WIN
+                startInfo.FileName = "netstat.exe";
+                startInfo.Arguments = "-a -n -o";
+#elif UNITY_EDITOR_OSX
+                startInfo.FileName = "netstat";
+                startInfo.Arguments = "-n -v -a";
+#endif
+                startInfo.WindowStyle = ProcessWindowStyle.Hidden;
+                startInfo.UseShellExecute = false;
+                startInfo.RedirectStandardInput = true;
+                startInfo.RedirectStandardOutput = true;
+                startInfo.RedirectStandardError = true;
+
+                using (var proc = new Process())
+                {
+                    proc.StartInfo = startInfo;
+                    proc.Start();
+#if UNITY_EDITOR_OSX
+                    proc.WaitForExit(2500);
+#endif
+
+                    var standardOutput = proc.StandardOutput;
+                    var standardError = proc.StandardError;
+
+                    var netStatContent = standardOutput.ReadToEnd() + standardError.ReadToEnd();
+                    var netStatExitStatus = proc.ExitCode.ToString();
+
+                    if (netStatExitStatus != "0")
```

Comment:
If this happens, we should not continue executing this method right? 
Also, what is the best course of action for the user? Is it possible to make the log more informative?